src/util/mapHelpers.js

```diff
@@ -1,81 +1,107 @@
 import d3 from "d3";
-import {range, isFinite} from "lodash";
-import linspace from "linspace";
-import outerProducts from "outer-product";
+import _ from "lodash";
 
 /* util */
 
-const Bernstein = (n, k) => {
-  var binom = require('binomial')
-  const coeff = binom.get(n, k) // calculate binomial coefficient
+export const pathStringGenerator = d3.svg.line()
+  .x((d) => { return d.x })
+  .y((d) => { return d.y })
+  .interpolate("basis");
+
+const extractLineSegmentForAnimationEffect = (
+  numDateMin,
+  numDateMax,
+  originCoords,
+  destinationCoords,
+  originNumDate,
+  destinationNumDate,
+  visible,
+  bezierCurve,
+  bezierDates
+) => {
 
-  const _bpoly = (x) => {
-    return coeff * Math.pow(x,k) * Math.pow(1-x,n-k)
+  if (visible === "hidden") {
+    return [];
   }
-  return _bpoly //return Bernstein polynomial
-}
 
-function zeros(dimensions) {
-    var array = [];
+  // want to slice out all points that lie between numDateMin and numDateMax
+  // and append interpolated start and end points
+  // full data
+  // bezierDates = [ 2015.1    2015.2    2015.3    2015.4    2015.5 ]
+  // bezierCurve = [ x1,y1     x2,y2     x3,y13    x4,y4     x5,y5  ]
+  // slice via numDateMin = 2015.25, numDateMax = 2015.45
+  // bezierDates = [ 2015.25  2015.3    2015.4  2015.45 ]
+  // bezierCurve = [ x23,y23  x3,y13    x4,y4   x45,y45  ]
+
+  // find start
+  const startIndex = _.findIndex(bezierDates, function (d) { return d > numDateMin });
+
+  // find end
+  const endIndex = _.findLastIndex(bezierDates, function (d) { return d < numDateMax });
+
+  // startIndex and endIndex is -1 if not found
+  // this indicates a slice of time that lies outside the bounds of BCurve
+  // return empty array
+  if (startIndex == -1 || endIndex == -1) {
+    return [];
```

Comment:
Is this right? What if part of the transmission were in the time slice, yet the max date is after the transmission? (or vice versa)